kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/MctsBeneficiaryImportServiceImpl.java

```diff
@@ -0,0 +1,490 @@
+        BufferedReader bufferedReader = new BufferedReader(reader);
+        readHeader(bufferedReader); // ignoring header as all interesting data in tab separated rows
+
+        CsvMapImporter csvImporter = new CsvImporterBuilder()
+                .setProcessorMapping(getChildProcessorMapping())
+                .setPreferences(CsvPreference.TAB_PREFERENCE)
+                .createAndOpen(bufferedReader);
+
+        try {
+            Map<String, Object> record;
+            while (null != (record = csvImporter.read())) {
+                importChildRecord(record);
+            }
+        } catch (ConstraintViolationException e) {
+            throw new CsvImportDataException(String.format("MCTS child import error, constraints violated: %s",
+                    ConstraintViolationUtils.toString(e.getConstraintViolations())), e);
+        }
+    }
+
+    private void importMotherRecord(Map<String, Object> record) {
+        MctsMother mother = (MctsMother) record.get(BENEFICIARY_ID);
+        String name = (String) record.get(BENEFICIARY_NAME);
+        Long msisdn = (Long) record.get(MSISDN);
+        DateTime lmp = (DateTime) record.get(LMP);
+
+        String errors = setLocationFields(record, mother);
+        if (errors != null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.INVALID_LOCATION, SubscriptionPackType.PREGNANCY);
+            return;
+        }
+
+        if (lmp == null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.MISSING_LMP, SubscriptionPackType.PREGNANCY);
+            return;
+        }
+
+        // TODO: more data validation specified in #111
+        mother.setName(name);
+
+        processSubscriptionForBeneficiary(mother, msisdn, lmp, SubscriptionPackType.PREGNANCY);
+    }
+
+
+    private void importChildRecord(Map<String, Object> record) {
+        MctsChild child = (MctsChild) record.get(BENEFICIARY_ID);
+        String name = (String) record.get(BENEFICIARY_NAME);
+        Long msisdn = (Long) record.get(MSISDN);
+        MctsMother mother = (MctsMother) record.get(MOTHER_ID);
+        DateTime dob = (DateTime) record.get(DOB);
+
+        String errors = setLocationFields(record, child);
+        if (errors != null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.INVALID_LOCATION, SubscriptionPackType.CHILD);
+            return;
+        }
+
+        if (dob == null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.MISSING_DOB, SubscriptionPackType.CHILD);
+            return;
+        }
+        // TODO: more data validation specified in #111
+
+        child.setName(name);
+        child.setMother(mother);
+
+        processSubscriptionForBeneficiary(child, msisdn, dob, SubscriptionPackType.CHILD);
+    }
+
+
+    private void processSubscriptionForBeneficiary(MctsBeneficiary beneficiary, Long msisdn, DateTime referenceDate,
+                                                  SubscriptionPackType type) {
```

Comment:
super nitpick, but can you consider renaming this to packType instead of just type? it isn't super obvious as to what is being passed into other method below.